doc/contrib/recording.md

```diff
@@ -0,0 +1,82 @@
+# Session Recording Technical Documentation
+
+The purpose of this document is to explain in detail the concepts and implementation of Elektra's session recording feature.
+While the feature is under development, it acts as sort of in-between documentation, roadmap and TODO list.
+
+## What is Session Recording?
+
+After every `kdbSet` action, the changes are calculated using Elektra's powerful changetracking API.
+The result of the calculation is an `ElektraDiff` instance we'll call _part diff_ throughout this document.
+The session recording plugin merges those _part diffs_ together and creates and persists an overall _session diff_.
+Importantly, the _session diff_ is persisted cross-process.
+Conceptually, this is depicted in the following image.
+
+![Session vs Part diff](../images/elektra-record/recording-part-diff.svg)
```

Comment:
>  I guess it would as if I called kdb record-stop and kdb record-start around the edit.

Exactly.

> Also: Maybe check whether your diagrams can be drawn with [Mermaid](https://mermaid.js.org/intro/).

I originally intended to do that, but the results were pretty unusable. The Session v Part Diff diagram was not too bad, bad the keystate diagram was too difficult for mermaid.